src/coreclr/src/dlls/mscoree/mscoree.cpp

```diff
@@ -327,7 +287,15 @@ HRESULT SetInternalSystemDirectory()
             // use local buffer for thread safety
             PathString wzSystemDirectory;
 
-            hr = GetCORSystemDirectoryInternaL(wzSystemDirectory);
+#ifdef TARGET_UNIX
+
+            // REVIEW: If I do not do this, I get linker errors about PAL_Random missing. Why?
```

Comment:
Looks like traditional Unix linker ordering problem. A better way to fix that maybe to put palrt inside the whole-archive section in the CMakeList.txt.

Reply:
Or inside `START_LIBRARY_GROUP`

Reply:
`START_LIBRARY_GROUP` seems like fixes this (at least on Ubuntu). Thanks!